tests/run/parallel.pyx

```diff
@@ -32,7 +32,7 @@ def test_parallel():
 
     free(buf)
 
-cdef int get_num_threads() with gil:
+cdef int get_num_threads() noexcept with gil:
```

Comment:
I believe it's also `except -1 with gil` but `nogil except -1`, for instance, so the current placement of `noexcept` is consistent with that.

Do we want to change this?

Reply:
Counts as deliberate breakage, I guess. I think the way forward is then to allow both placements for one(!) of the two modifiers already in 0.29.x, and then deprecate one of them by first making the documentation consistent.

Since `with gil`reads best at the end of the signature line (because it's similar to a `with gil:` statement inside of the function), I think we should allow `except -1 nogil`, and then newly require `noexcept nogil`, so that people get used to that order.